src/ucp/core/ucp_worker.c

```diff
@@ -2524,28 +2526,40 @@ static void ucp_worker_destroy_eps(ucp_worker_h worker,
     }
 }
 
+static void ucp_worker_eps_cleanup(ucp_worker_h worker)
+{
+    /* Cleanup hash of discarded UCT EPs prior destroying all UCP EPs to
+     * destroy UCP EPs which were already marked as closed by a user, but had
+     * discarding is in progress */
+    ucp_worker_discard_uct_ep_cleanup(worker);
```

Comment:
do we need to call it here to avoid destroying EPs which have ongoing discard procedure?

Reply:
yes, right

Reply:
how can progress happen before line 2540? Why would we need to invoke it here?

Reply:
we do it from here - https://github.com/openucx/ucx/blob/66f8240670c298f29dd3379e793611927b979a5f/src/ucp/wireup/wireup_ep.c#L334